provider/slof.py

```diff
@@ -0,0 +1,192 @@
+"""
+Module for providing common interface to test SLOF component.
+"""
+
+import re
+import time
+import logging
+import os
+
+from virttest import utils_misc
+
+
+class _LazyReader(object):
```

Comment:
`_LazyReader` could open console log once for all if you use `self._fd.seek(0)` in `__iter__` and `_get_len`.
Like:
```python
class _LazyReader(object):
    def __init__(self, vm):
        self.fd = open(vm.serial_console_log)

    def __iter__(self):
        pos = 0
        self.fd.seek(pos)
        ...

    @property
    def len(self):
        self.fd.seek(0)
        ...

    def __getitem__(self, index):
        if isinstance(index, slice):
            start, stop, step = index.indices(self.len)
            ...
```

Reply:
nice one :+1: